### PR TITLE
Update cross-compilation docs for Raspberry Pi 4 (gh #1093)

### PR DESCRIPTION
**Pull request by ewertons**
Addresses issue https://github.com/Azure/azure-iot-sdk-c/issues/1093

 

# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 
  - [x] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)
https://github.com/Azure/azure-iot-sdk-c/issues/1093

# Description of the problem

# Description of the solution

### CODE REVIEW

File: doc/SDK_cross_compile_example.md [hunk 1]
```diff
@@ -108,12 +112,20 @@ If you need to provide additional build flags for your cross compile to function
...
 ### Notes
-These instructions have been tested on both the Raspberry Pi 2 and 3.
+These instructions have been tested on both the Raspberry Pi 2, 3 and 4.
 
 ### Known Issues and Circumventions
 
 If you encounter the error _error adding symbols: DSO missing from command line_ try adding a reference to libdl with  _-cl -ldl_ added to your build script command line.
 
+For Raspberry Pi4 the following workaround is neccessary to properly link the samples with pthread.
+Run the following commands and [recompile the Azure IoT C SDK](#Building-the-SDK).
```
Comment (danewalton) [inline doc/SDK_cross_compile_example.md:122]: nit: maybe provide an explanation for what the command is doing?

File: doc/SDK_cross_compile_example.md [hunk 2]
```diff
@@ -108,12 +112,20 @@ If you need to provide additional build flags for your cross compile to function
...
 
 ### Notes
-These instructions have been tested on both the Raspberry Pi 2 and 3.
+These instructions have been tested on both the Raspberry Pi 2, 3 and 4.
 
 ### Known Issues and Circumventions
 
 If you encounter the error _error adding symbols: DSO missing from command line_ try adding a reference to libdl with  _-cl -ldl_ added to your build script command line.
 
+For Raspberry Pi4 the following workaround is neccessary to properly link the samples with pthread.
```
Comment (danewalton) [inline doc/SDK_cross_compile_example.md:121]:
```suggestion
For Raspberry Pi 4 the following workaround is neccessary to properly link the samples with pthread.
```

**Review by danewalton — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

File: doc/SDK_cross_compile_example.md [hunk 3]
```diff
@@ -108,12 +112,20 @@ If you need to provide additional build flags for your cross compile to function
...
+Run the following commands and [recompile the Azure IoT C SDK](#Building-the-SDK).
+
+```bash
+cd $RPI_ROOT/usr/lib/arm-linux-gnueabihf; 
+ln -s ../../../lib/arm-linux-gnueabihf/libpthread.so.0 libpthread.so
```
Comment (CIPop) [inline doc/SDK_cross_compile_example.md:126]: Is this a bug on RPI4's distro or are we referencing the pthread lib incorrectly?

Comment (ewertons) [reply doc/SDK_cross_compile_example.md:126]: I would say it's a problem with the new RPi OS. This same code of ours compiled without issues on Pi 3 (Raspbian).

File: doc/SDK_cross_compile_example.md (hunk 1 above)
Comment (ewertons) [reply doc/SDK_cross_compile_example.md:122]: I would say the issue that is linked gives information on that. I'm trying to keep the text small (or at least not growing it too much).

**Review by wduraes — APPROVED** [review]

**Pull request merged**